common/src/main/java/org/conscrypt/Conscrypt.java

```diff
@@ -81,6 +81,23 @@ public static Provider newProvider(String providerName) {
         return new OpenSSLProvider(providerName);
     }
 
+    /**
+     * Constructs a new {@link Provider} with the default name that also provides an implementation
+     * of {@link javax.net.ssl.TrustManagerFactory}.
+     */
+    public static Provider newProviderWithTrustManager() {
+        return newProviderWithTrustManager(Platform.getDefaultProviderName());
+    }
+
+    /**
+     * Constructs a new {@link Provider} with the given name that also provides an implementation
+     * of {@link javax.net.ssl.TrustManagerFactory}.
+     */
+    public static Provider newProviderWithTrustManager(String providerName) {
```

Comment:
why not always always include a trust manager?

Reply:
There are a few implementation differences between our TrustManager and the OpenJDK-provided one, so I didn't want to swap our users' TrustManager out from under them.  Users generally install Conscrypt as the first provider (so the OpenJDK SSL implementation isn't used), and if you don't specify a TrustManager (which most callers don't) the highest-priority one is automatically used, so providing it to everyone would cause them to verify certs differently, which might be a problem.

Maybe that's worth it, though.  I think our TrustManager impl is a good one, so it should be okay, but it might mean some certs fail to verify that previously did (or vice versa), and I was worried about springing that on people unexpectedly.  I'd be interested if you have an opinion.

Reply:
What if our implementation was controlled by some sort of flag? Maybe a system property?

Reply:
@nmittler I personally dislike using system properties and the like for things like this, since it can easily result in a tenuous connection between cause and effect, but I realize that may be a minority view.  I'd rather it be reasonably clear looking just at the provider install site what the call will do.

@chadbrubaker Yeah, if we included it by default we'd want to only do that on OpenJDK and leave it out on Android.  Users can trivially bypass the network security config TrustManager if they want (by just providing a TrustManager to SSLContext.init()), so I'm not sure it's worth preventing them from adding a new TrustManager entirely if they ask for it, but I definitely don't want to do it by default.

Reply:
@flooey do you ever envision a case where a user would want to have providers with/without trust manager on the same environment? If not, maybe we just add a single method to this API:

```java
public static void enableTrustManager(boolean enable) {
...
}
```

Reply:
@nmittler They're very unlikely to want to have more than one instance.  Maybe we should make it a builder?  Then we can easily slot in any future configuration options without much fuss.

```java
Conscrypt.newProviderBuilder().setName("Foo").provideTrustManager().build()
```

Reply:
That seems reasonable.  You'll need to mark the old methods as deprecated.

Reply:
@flooey Agreed, we don't (and don't want to) try and prevent devs from replacing our provider on purpose, I was just worried about it happening unexpectedly when they used the unbundled provider installer.